Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -1369,7 +1369,7 @@ checkCabalVersion pkg =
       _                   -> False
 
     versionRangeExpressions =
-        [ dep | dep@(Dependency _ vr) <- buildDepends pkg
+        [ dep | dep@(Dependency _ vr) <- allBuildDepends pkg
```

Comment:
This makes my BC senses tingle. What's the BC story for custom setups that call `buildDepends`. Can we preserve the old semantics under the old name?

Reply:
https://github.com/23Skidoo/all-custom-setups says `buildDepends` it isn't really used anywhere, `pandoc-1.6` is from 2010

```
~/all-custom-setups$ git grep   -e buildDepends origin/all-custom-setups
origin/all-custom-setups:open-pandoc-1.5.1.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.5.0.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.5.1.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.5.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.5/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.6.0.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.6/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
ogre@bigtest:~/all-custom-setups$ git grep   -e buildDepends origin/all-custom-setups
origin/all-custom-setups:open-pandoc-1.5.1.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.5.0.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.5.1.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.5.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.5/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.6.0.1/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
origin/all-custom-setups:pandoc-1.6/Setup.hs:  let highlightingSupport = any isHighlightingKate $ buildDepends pkg
```

Reply:
Yeah per the big comment at https://github.com/haskell/cabal/pull/4383/files#diff-d02dbf8d6911c6afb8c8cb25ebc6ea9cL127, anything that uses it seems bound to be incorrect anyways.